### PR TITLE
GithubAPI Groovy Script conversion

### PR DESCRIPTION
**Pull request by TheMeinerLP**
In this PR I have converted the GithubAPI script into a Java class. I have reduced code and implemented better error handling.
API and implementation are separated and can be called via Singelton. 

Overall, this PR should work simply because the classes and calls have the same names

### CODE REVIEW

**Comment by TheMeinerLP** [comment]
Waiting for #3936 get merged

**Comment by TheMeinerLP** [comment]
Tests needs be added

**Review by timja — COMMENTED** [review]
I think the code would be simplified if try catches weren't added everywhere, apart from that it would be good to use brand names appropriately throughout.

It is GitHub, not Github or github

File: src/main/java/io/jenkins/infra/repository_permissions_updater/GitHubImpl.java
Comment (TheMeinerLP) [reply src/main/java/io/jenkins/infra/repository_permissions_updater/GitHubImpl.java:74]: Roger that

File: src/test/java/io/jenkins/infra/repository_permissions_updater/helper/URLHelper.java
Comment (timja) [inline src/test/java/io/jenkins/infra/repository_permissions_updater/helper/URLHelper.java:12]: this class and a bunch of other mocks could be avoided by using https://wiremock.org/ instead of mocking the inputstream responses.

Wiremock would also mean less re-work on tests required if someone gets around to modernising the http code.

Not a blocker just a suggestion if you have time (I'll merge this now, but if you want to do a follow up that would be appreciated).

**Review by timja — APPROVED** [review]

File: src/test/java/io/jenkins/infra/repository_permissions_updater/helper/URLHelper.java
Comment (TheMeinerLP) [reply src/test/java/io/jenkins/infra/repository_permissions_updater/helper/URLHelper.java:12]: I would rather do it as downstream PR

**Pull request merged**
